todo_app/models/task.py

```diff
@@ -0,0 +1,46 @@
+from todo_app.models.label import Label
+
+
+class Task:
+  def __init__(self, id, title, labels):
+    self.id = id
+    self.title = title
+    self.labels = labels
+    self.status = self.find_status()
+
+  @classmethod
+  def from_trello_card(cls, card):
+    labels = [Label.from_trello_label(t_label) for t_label in card["labels"]]
+    return cls(card['id'], card['name'], labels)
+
+  def find_status(self):
+    status_options = ['Not Started', 'In Progress', 'Complete']
+    statuses = [label for label in self.labels if label.title in status_options]
+    status_count = len(statuses)
+    if status_count == 1:
+        return statuses[0]
+    elif status_count < 1:
+        return None
+    else:
+        return statuses
```

Comment:
While Python is completely happy with it, I'd still avoid having a method returning different types where you can - it's not easy for calling code to decide whether to expect a List or a string here and they need to account for both every time

Particularly in this structure, the fact Trello isn't enforcing our model constraints is becoming very leaky here and forcing other code to think carefully about all the possibilities

For example, what if we adjusted this to be:
```python
  def find_status(self):
    status_options = ['Not Started', 'In Progress', 'Complete']
    statuses = [label for label in self.labels if label.title in status_options]
    status_count = len(statuses)
    if status_count == 1:
        return statuses[0]
    elif status_count < 1:
        return Label(None, "No Status")
    else:
        return Label(None, "MultiStatus")
```
That way every combo should produce the same Label type, which also simplifies our checks below, e.g.
```python
  def not_started(self):
    return self.status.title == "Not Started"
```
With a small adjustment to the HTML to loop over the labels rather than the statuses for a MultiStatus to-do, I think that follows all the current expectations

Reply:
rad idea.
It will require some small rejig to what I do with tasks with multiple stati, but that whole area is also gross.